doc/classes/WorkerThreadPool.xml

```diff
@@ -1,10 +1,50 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="WorkerThreadPool" inherits="Object" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
+		A singleton that allocates some [Thread]s on startup, used to offload tasks to these threads.
 	</brief_description>
 	<description>
+		The [WorkerThreadPool] singleton allocates a set of [Thread]s (called worker threads) on project startup and provides methods for offloading tasks to them. This can be used for simple multithreading without having to create [Thread]s.
+		Tasks hold the [Callable] to be run by the threads. [WorkerThreadPool] can be used to create regular tasks, which will be taken by one worker thread, or group tasks, which can be distributed between multiple worker threads. Group tasks execute the [Callable] multiple times, which makes them useful for iterating over a lot of elements, such as the enemies in an arena.
+		Here's a sample on how to offload an expensive function to worker threads:
+		[codeblocks]
+		[gdscript]
+		var enemies = [] # An array to be filled with enemies.
+
+		func process_enemy_ai(enemy_index):
+		    var processed_enemy = enemies[enemy_index]
+		    # Expensive logic...
+
+		func _process(delta):
+		    var task_id = WorkerThreadPool.add_group_task(process_enemy_ai, enemies.size())
+		    # Other code...
+		    WorkerThreadPool.wait_for_group_task_completion(task_id)
+		    # Other code that depends on the enemy AI already being processed.
+		[/gdscript]
+		[csharp]
+		private Node[] _enemies = {}; // An array to be filled with enemies.
```

Comment:
Since C# arrays are immutable, I think it'd be better to use `List` here. Then, in `_Process` replace `enemies.Length` with `enemies.Count`.

It probably doesn't matter too much, but I think using a mutable collection would be better for a code example because then the user can change its size which the comment seems to suggest. Also, C# lists are more similar to GDScript's arrays than C# arrays. Sorry for making you change this again :disappointed:.

```suggestion
		private List&lt;Node&gt; _enemies = new List&lt;Node&gt;(); // An array to be filled with enemies.
```

Reply:
So "A list to be filled with enemies" then?